tests/orion/schemas/test_schedules.py

```diff
@@ -647,6 +647,52 @@ async def test_rrule_validates_rrule_str(self):
         with pytest.raises(ValidationError, match="(invalid 'FREQ': DAILYBAD)"):
             RRuleSchedule(rrule="FREQ=DAILYBAD")
 
+    async def test_rrule_schedule_handles_complex_rrulesets(self):
+        s = RRuleSchedule(
+            rrule="DTSTART:19970902T090000\n"
+            "RRULE:FREQ=YEARLY;COUNT=2;BYDAY=TU\n"
+            "RRULE:FREQ=YEARLY;COUNT=1;BYDAY=TH\n"
+        )
+        dates_from_1900 = await s.get_dates(5, start=datetime(1900, 1, 1, tz="UTC"))
+        dates_from_2000 = await s.get_dates(5, start=datetime(2000, 1, 1, tz="UTC"))
+        assert len(dates_from_1900) == 3
+        assert len(dates_from_2000) == 0
+
+    @pytest.mark.xfail(reason="we currently cannot roundtrip RRuleSchedule objects")
+    async def test_rrule_schedule_handles_rruleset_roundtrips(self):
+        s1 = RRuleSchedule(
+            rrule="DTSTART:19970902T090000\n"
+            "RRULE:FREQ=YEARLY;COUNT=2;BYDAY=TU\n"
+            "RRULE:FREQ=YEARLY;COUNT=1;BYDAY=TH\n"
+        )
+        s2 = RRuleSchedule.from_rrule(s1.to_rrule())
+        s1_dates = await s1.get_dates(5, start=datetime(1900, 1, 1, tz="UTC"))
+        s2_dates = await s2.get_dates(5, start=datetime(1900, 1, 1, tz="UTC"))
+        assert s1_dates == s2_dates
+
+    async def test_rrule_schedule_rejects_rrulesets_with_many_dtstarts(self):
+        dt_nyc = datetime(2018, 1, 11, 4, tz="America/New_York")
+        dt_chicago = datetime(2018, 1, 11, 3, tz="America/Chicago")
+        rrset = rrule.rruleset(cache=True)
+        rrset.rrule(rrule.rrule(rrule.HOURLY, count=10, dtstart=dt_nyc))
+        rrset.rrule(rrule.rrule(rrule.HOURLY, count=10, dtstart=dt_chicago))
+
+        with pytest.raises(ValueError):
```

Comment:
Can we add the `match="exception message"` part to this assertion so it's clear which of the two exception conditions we're testing?